### PR TITLE
Add POST method support

### PR DESCRIPTION
**Pull request by ykitamura-mdsol**
Modify to be able to handle POST method

@mdsol/team-10 please review

### CODE REVIEW

**Comment by jcarres-mdsol** [comment]
LGTM. I do not know if it is difficult to write a spec for this or the project in general lacks

**Comment by HonoreDB** [comment]
Sorry, would you mind explaining why this is necessary? Why can't you just pass the body in using attributes and leave params empty on a POST? This looks like it makes it impossible to make a POST containing params, which is unusual but not unheard of.

File: lib/farscape/transition.rb [hunk 1]
```diff
@@ -23,8 +23,12 @@ def invoke(args={})
       call_options[:url] = @transition.uri
       call_options[:method] = @transition.interface_method
       call_options[:headers] = @agent.get_accept_header(@agent.media_type).merge(options.headers || {})
-      call_options[:params] = options.parameters unless options.parameters.blank?
       call_options[:body] = options.attributes unless options.attributes.blank?
+      if @transition.interface_method.to_s.downcase == 'get'
+        call_options[:params] = options.parameters unless options.parameters.blank?
```
Comment (jcarres-mdsol) [inline lib/farscape/transition.rb:38]: @HonoreDB  I see, you you are saying this line stays where it used to be

File: lib/farscape/transition.rb [hunk 2]
```diff
@@ -23,8 +23,12 @@ def invoke(args={})
...
+      else
+        call_options[:body] = (call_options[:body] || {}).merge(options.parameters) unless options.parameters.blank?
```
Comment (jcarres-mdsol) [inline lib/farscape/transition.rb:40]: And we only add this line?

File: lib/farscape/transition.rb (hunk 1 above)
Comment (HonoreDB) [inline lib/farscape/transition.rb:38]: Yes.

File: lib/farscape/transition.rb (hunk 2 above)
Comment (HonoreDB) [inline lib/farscape/transition.rb:40]: I'm not clear on why we need this line. Why can't you just pass the body in using attributes and leave params empty on a POST?

Comment (jcarres-mdsol) [inline lib/farscape/transition.rb:40]: @ykitamura-mdsol  I've talked to @HonoreDB  and he has suggested to add here a big TODO explaining why we need to do this , maybe how to trigger it so we remember what was this for when we come here again in the future

**Comment by ykitamura-mdsol** [comment]
I have added a TODO and specs

@jcarres-mdsol  @HonoreDB  please review

**Comment by HonoreDB** [comment]
Thanks, looks good to me.

File: lib/farscape/transition.rb [hunk 3]
```diff
@@ -23,9 +23,23 @@ def invoke(args={})
...
-      call_options[:params] = options.parameters unless options.parameters.blank?
       call_options[:body] = options.attributes unless options.attributes.blank?
 
+      # TODO: Farscape handles "parameters" as query string, and "attributes" as request body.
+      # However, in many API documents, only "parameters" is used regardless of methods.
+      # Here is an example of POST method using "parameters".
+      # https://github.com/mdsol/Archon/blob/436cd06de7f61d49390a82a4be6f2aca799ec277/apis/api_document_v1.yml#L308
+      #
+      # Since changing API documents must have a huge impact on existing systems,
+      # modified Farscape to use "parametes" as request body if method is not GET.
```
Comment (ssteeg-mdsol) [inline lib/farscape/transition.rb:34]: Sorry for the late comment, but 
"parametes" → "parameters"

File: lib/farscape/transition.rb [hunk 4]
```diff
@@ -23,9 +23,23 @@ def invoke(args={})
...
+      #
+      # This modification makes impossible to make a POST containing params.
```
Comment (ssteeg-mdsol) [inline lib/farscape/transition.rb:36]: This modification makes impossible to make a POST containing params.

This modification means it is impossible to make a POST containing params. \*

**Pull request merged**